chapter_1/exercise_1_22/fold_line.c

```diff
@@ -43,30 +43,43 @@ int get_line(char line[], int max_line_len)
 
 void fold_line(char line[], char fold_str[], int n_break)
 {
-  int i, j;
+  int i, j, m;
+  int colon = 0;
   int split = FALSE;
-  int offset = 10;
 
-  for (i = 0, j = 0; line[i] != '\0'; ++i, ++j)
+  for (i = 0, j = 0; line[i] != '\0'; i++, j++)
   {
-    if (i && i % n_break == 0)
-    {
-      split = TRUE;
-    }
+    fold_str[j] = line[i];
 
-    if (split && (line[i] == ' ' || i > n_break + offset))
+    if (fold_str[j] == '\n')
     {
-      fold_str[j++] = '\n';
-      split = FALSE;
-      ++i;
+      colon = 0;
     }
 
-    fold_str[j] = line[i];
-  }
+    colon++;
 
-  if (line[i] == '\n')
-  {
-    fold_str[j++] = line[i];
+    if (colon == n_break)
+    {
+      split = TRUE;
+      for (m = j; m >= j - 10 && split; m--)

```

Comment:
@illustrofia Why did you put the magic number "10" in here?

Reply:
It was supposed to replace offset. I shall replace it with a variable so that the code is more clear.

Reply:
What do you mean by offset? Can you elaborate a little bit on this?

Reply:
What I am here referring to as 'offset' is the number of columns we search for a whitespace character to replace it with a newline character.